openslides/agenda/static/js/agenda/site.js

```diff
@@ -268,13 +269,22 @@ angular.module('OpenSlidesApp.agenda.site', [
                     return false;
             }
         };
-        $scope.getUpdateStatePrefix = function (item) {
+        $scope.getDetailStatePrefix = function (item) {
             var prefix = item.content_object.collection.replace('/','.');
             // Hotfix for Issue 2566.
             // The changes could be reverted if Issue 2480 is closed.
             prefix = prefix.replace('motion-block', 'motionBlock');
             return prefix;
         };
+        $scope.edit = function (item) {
+            var formName = item.content_object.collection.split('/')[1];
+            // Hotfix for Issue 2566.
+            // The changes could be reverted if Issue 2480 is closed.
+            formName = formName.replace('motion-block', 'motionBlock');
+            formName = formName.charAt(0).toUpperCase() + formName.slice(1) + 'Form';
```

Comment:
Yes it is. The server transforms the python class name MotionBlock in the app motions to the collection name `motions/motion-block`. This name is send to the client. The resource definition in the client is `motions/motion-block`. Now you have to create a nice generic method that transforms `motions/motion-block` to `MotionBlock` + `Form` for this generic form dialog thing here. Can you please do it in #2983?